### DISCUSSION TITLE
Can't get to work custom colors and opacity modifiers

### DISCUSSION THREAD

**Discussion by lessless** [tailwindlabs/tailwindcss#8300] [Help]
Hi,

If I use opacity modifiers with builtin colors, then everything is ok - `<h1 class="text-gray-300 hover:text-gray-700 font-light  hover:font-bold text-5xl text-center">Index 001</h1>` works as expected. 

But if I'm trying to use the color specified in `tailwind.config.js` then it's not working - `<h1 class="text-primary-300 hover:text-primary-700 font-light  hover:font-bold text-5xl text-center">Index 001</h1>`

Here is my `tailwind.config.js`

```
module.exports = {
  content: ["./js/**/*.js", "../lib/*_web.ex", "../lib/*_web/**/*.*ex"],
  theme: {
    extend: { colors: { primary: "#5c6ac4" } },
  },
  plugins: [require("@tailwindcss/forms")],
};

```

I checked the resulting CSS file and it seems that Tailwind doesn't generate classes for that custom color, whilst for the default color I'm seeing 

```
.text-gray-300 {
  --tw-text-opacity: 1;
  color: rgb(209 213 219 / var(--tw-text-opacity));
}

.hover\:text-gray-700:hover {
  --tw-text-opacity: 1;
  color: rgb(55 65 81 / var(--tw-text-opacity));
}
```

What am I missing?

**Comment by aggreggator** [comment]
You have only defined primary and not primary-300 or primary-700.

**Comment by lessless** [comment]
@aggreggator I assumed that Tailwind would generate variants for me.
Maybe I missed this bit in the docs. 

Would you please help me define those - what's the format and what variations should I define in general?

**Reply by aggreggator** [reply]
> @aggreggator I assumed that Tailwind would generate variants for me. Maybe I missed this bit in the docs.

Tailwind can't know how to create a color palette.
Best help is to read the docs: https://tailwindcss.com/docs/customizing-colors#color-object-syntax

> Would you please help me define those - what's the format and what variations should I define in general?

There are some tailwind palette generators:
https://lmgtfy.app/#gsc.tab=0&gsc.q=tailwind%20color%20palette%20generator
